components/mediation/data-publishers/org.wso2.micro.integrator.prometheus.publisher/src/main/java/org/wso2/micro/integrator/prometheus/publisher/service/MetricResource.java

```diff
@@ -69,7 +77,13 @@ public boolean invoke(MessageContext synCtx) {
 
         if (metrics != null && !metrics.isEmpty()) {
             log.debug("Retrieving metric data successful");
-            textRootElem.setText(String.join("", metrics));
+            try {
+                StringBuilder sbr = metricFormatter.formatMetrics(registry.
+                                                                     filteredMetricFamilySamples(parseQuery(query)));
+                textRootElem.setText(sbr.toString());
+            } catch (IOException e) {
+                log.error("Error in parsing metrics.", e);
```

Comment:
I think we need to build an error response and reply? At least with an error code?

Reply:
I think there is no need to send the error created during Prometheus Metric creation to the client as it has little use to him. Instead of sending the error response to the client, the error generated here is sent to the  Prometheus Server where it's recognized as no data .